### PR TITLE
Change gpu_batch_predict release test benchmark

### PR DESCRIPTION
**Pull request by jiaodong**
## Why are these changes needed?

This should use latest numpy udf with numpy prediction now, also update single column selection code in new numpy predictor as it's not necessary.

Also this PR changes 
 - g3 compute template with g4dn 
 - re-enabled 100gb gpu prediction test -- we can now run it faster with less memory footprint !

```

4 x g4dn.12xlarge, 100GB data

'total_time_s', 'perf_metric_value': 304.01
'throughout_MB_s', 'perf_metric_value': 336.83102529522057

1 x g4dn.8xlarge, 20GB data

'total_time_s', 'perf_metric_value': 285.2
'throughout_MB_s', 'perf_metric_value': 71.80925666199158

1 x g4dn.8xlarge, 1GB data

'total_time_s', 'perf_metric_value': 46.12, 
'perf_metric_name': 'throughout_MB_s', 'perf_metric_value': 22.20294882914137
```

new runs on CI:

20GB image data https://buildkite.com/ray-project/release-tests-pr/builds/21577#01848692-1ff0-4ec5-bb83-7fdea0ca4e89
100GB image data https://buildkite.com/ray-project/release-tests-pr/builds/21577#01848692-1ff3-4f18-852f-9f660dbd1487

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [x] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-air/benchmarks.rst [hunk 1]
```diff
@@ -113,26 +113,26 @@ using a Pytorch pre-trained ResNet model.
...
+    * - 1 g4dn.8xlarge node
       - 1 GB (1623 images)
-      - 72.59 s (22.3 images/sec)
+      - 46.12 s (35.19 images/sec)
       - `python gpu_batch_prediction.py --data-size-gb=1`
-    * - 1 g3.8xlarge node
+    * - 1 g4dn.8xlarge node
       - 20 GB (32460 images)
-      - 1213.48 s (26.76 images/sec)
+      - 285.2 s (113.81 images/sec)
```
Comment (jiaodong) [inline doc/source/ray-air/benchmarks.rst:131]: g4dn provides decent speed up over g3 but this number shows we've come a long way in perf ;)

**Comment by xwjiang2010** [comment]
is the lint error relevant?

**Review by xwjiang2010 — APPROVED** [review]
Thanks Jiao! Release tests are passing. Pending clarifying the lint errors.

**Comment by jiaodong** [comment]
Screen Shot 2022-11-17 at 11 46 51 AM

it should be irrelevant, also fails on master

**Comment by xwjiang2010** [comment]
hmmm, the error has different signature than the master one tho:
```


BROKEN LINKS SUMMARY:
--
  |  
  | (ray-air/benchmarks: line  116) broken    https://github.com/ray-project/ray/blob/master/release/air_tests/air_benchmarks/compute_gpu_1_g4_8xl.yaml#L6-L15 - 404 Client Error: Not Found for url: https://github.com/ray-project/ray/blob/master/release/air_tests/air_benchmarks/compute_gpu_1_g4_8xl.yaml


```

**Review by amogkam — APPROVED** [review]
Excellent!

**Review by richardliaw — APPROVED** [review]

**Pull request merged**